content/developer/howtos/frontend_owl_components.rst

```diff
@@ -0,0 +1,187 @@
+files:
+
+.. code-block:: py
+
+    {
+        # ...
+        'assets': {
+            'web.assets_frontend': [
+                'your_module/static/src/portal_component/**/*',
+            ],
+        }
+    }
+
+.. seealso::
+    :ref:`Module manifest reference<reference/module/manifest>`.
+
+3. Adding an `<owl-component>` tag to a page
+============================================
+
+Now we need add an `<owl-component>` tag that will serve as the target for the component
+to be mounted. For the sake of this example, we'll add it directly to the portal's
+home page with an xpath in :file:`/your_module/views/templates.xml`.
+
+.. code-block:: xml
+
+    <?xml version="1.0" encoding="utf-8"?>
+    <odoo>
+        <template id="your_module.portal_my_home" inherit_id="portal.portal_my_home">
+            <xpath expr="//*[hasclass('o_portal_my_home')]" position="before">
+                <owl-component name="your_module.YourComponent" />
+            </xpath>
+        </template>
+    </odoo>
+
+Don't forget to add this file to the data section of your assets bundle:
+
+.. code-block:: py
+
+    {
+        # ...
+        'data': {
+            'views/templates.xml',
+        }
+    }
+
+And that's it! If you open the home page of the portal you should see the message
+"Hello, World!" at the top of the page.
+
+Problems with using Owl components in the portal and website
+============================================================
+
+Owl components are rendered entirely in JavaScript by the browser. This can cause
+some issues:
+
+- Layout shift
+- Poorer indexing by search engines
+
+For these reasons, you should only use Owl components in the portal and website for
+specific use cases described below.
+
+Layout shift
+------------
+
+When a page initially renders content, and that content subsequently moves ("shifts")
+within the page, this is refered to as a layout shift. When using Owl components in
+the portal or website, all of the HTML that surrounds the Owl component is rendered
+by the server, and is the first thing that will be displayed to the user. When JavaScript
+starts running, Owl will mount your component, which will likely cause the surrounding
+elements to move around in the page. This can cause poor user experience: the user sees
+and element in the page that was initially rendered and with which they want to interact,
+so they move their cursor or finger above that element. Just as they're about to click,
+the Owl component is mounted and the element with which they wanted to interact moved.
+They click and interact with the Owl app instead.
+
+This can be a frustrating experience, so you should be careful when designing your
+page that the Owl component will not move elements around. This can be achieved in
+various ways, e.g. by positioning it below all other existing elements, by not having
+other interactive elements around, or by reserving a fixed space for the Owl component
+using CSS.
```

Comment:
```suggestion
When a page initially renders content, and that content subsequently moves ("shifts")
within the page, this is referred to as a layout shift. When using Owl components in
the portal or website, all of the HTML that surrounds the Owl component is rendered
by the server and is the first thing that will be displayed to the user. When JavaScript
starts running, Owl will mount your component, which will likely cause the surrounding
elements to move around on the page. This can cause poor user experience: the user sees
an element on the page that was initially rendered and with which they want to interact,
so they move their cursor or finger above that element. Just as they're about to click,
the Owl component is mounted, and the element they want to interact with is moved.
They click and interact with the Owl app instead.

This can be a frustrating experience, so you should be careful when designing your
page that the Owl component will not move elements around. This can be achieved in
various ways, e.g. by positioning it below all other existing elements, not having
other interactive elements around, or reserving a fixed space for the Owl component
using CSS.
```